relayer/src/chain/cosmos.rs

```diff
@@ -655,6 +651,34 @@ impl Chain for CosmosSDKChain {
         }
         Ok(result)
     }
+
+    fn query_connection_channels(
+        &self,
+        request: QueryConnectionChannelsRequest,
+    ) -> Result<Vec<ChannelId>, Error> {
+        let grpc_addr =
+            Uri::from_str(&self.config().grpc_addr).map_err(|e| Kind::Grpc.context(e))?;
+        let mut client = self
+            .block_on(
+                ibc_proto::ibc::core::channel::v1::query_client::QueryClient::connect(grpc_addr),
+            )?
+            .map_err(|e| Kind::Grpc.context(e))?;
+
+        let request = tonic::Request::new(request);
+
+        let response = self
+            .block_on(client.connection_channels(request))?
+            .map_err(|e| Kind::Grpc.context(e))?
+            .into_inner();
+
+        let vec_ids = response
+            .channels
+            .iter()
+            .filter_map(|ic| ChannelId::from_str(ic.channel_id.as_str()).ok())
```

Comment:
If there are any errors here, they will be silently ignored. Maybe we can instead do something like this: https://doc.rust-lang.org/rust-by-example/error/iter_result.html#fail-the-entire-operation-with-collect

Reply:
We can do this but the question is do we want to fail the entire operation if we cannot extract one  `ChannelId`?

Reply:
Good catch.

Between the option of failing the operation because an identifier fails to parse, and ignoring the offending identifier(s), I would rather go with the latter.

One middle-road solution is to have warnings in case some identifiers fail to parse. We're not (yet) principled with how we use warnings, if at all, so I'm adding a TODO here to keep an eye out in the future. c51b814